tests/test_deploy_github_pages.py

```diff
@@ -0,0 +1,277 @@
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+        target_branch_exists = run(["git", "show-branch", f"remotes/origin/{target_branch}"], capture_output=True,
+                                   text=True)
+        assert target_branch_exists.returncode == 0
+        remove_branch(target_branch)
+
+
+def test_pushing_to_existing_docu_branch_same_source():
+    with TemporaryDirectory() as tempdir:
+        os.chdir(tempdir)
+        user_name, user_access_token = setup_test_repo()
+        doc_dir = run(["git", "rev-parse", "--show-toplevel"], capture_output=True, text=True, check=True)
+        os.chdir(f"{doc_dir.stdout[:-1]}/doc")
+        source_branch = "5-add-tests"
+        run(["git", "checkout", source_branch], check=True)
+        target_branch = "test-docu-new-branch"
+        remove_branch(target_branch)
+
+        cwd = os.getcwd()
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", source_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+        current_commit_id = run(["git", "ls-remote",
+                                 f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git",
+                                 target_branch], capture_output=True, check=True)
+        commit_id_old = current_commit_id.stdout
+
+        # make a change in the docu
+        with open("./index.rst", "a") as file:
+            file.write("\n\nThis text is a change.")
+
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", source_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+        current_commit_id = run(["git", "ls-remote",
+                                 f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git",
+                                 target_branch], capture_output=True, check=True)
+        commit_id_new = current_commit_id.stdout
+
+        # the docu files where updated, so a new commit should be pushed to the remote
+        assert not commit_id_new == ""
+        assert not commit_id_old == ""
+        assert not commit_id_old == commit_id_new
```

Comment:
Hm, this is a bit confusing.
you do a change in the docu, but don't commit it.
Then you have this comment:
# the docu files where updated, so a new commit should be pushed to the remote

But actually you check:
`assert not commit_id_old == commit_id_new`
?